yt_dlp/extractor/thisoldhouse.py

```diff
@@ -39,17 +55,63 @@ class ThisOldHouseIE(InfoExtractor):
         'url': 'https://www.thisoldhouse.com/21083431/seaside-transformation-the-westerly-project',
         'only_matching': True,
     }]
-    _ZYPE_TMPL = 'https://player.zype.com/embed/%s.html?api_key=hsOk_yMSPYNrT22e9pu8hihLXjaZf0JW5jsOWv4ZqyHJFvkJn6rtToHl09tbbsbe'
+
+    _LOGIN_URL = 'https://login.thisoldhouse.com/usernamepassword/login'
+
+    def _perform_login(self, username, password):
+        self._request_webpage(
+            HEADRequest('https://www.thisoldhouse.com/insider'), None, 'Requesting session cookies')
+
+        urlh = self._request_webpage(
+            'https://www.thisoldhouse.com/wp-login.php', None, 'Requesting login info',
+            errnote='Unable to login', query={'redirect_to': 'https://www.thisoldhouse.com/insider'})
+        login_info = traverse_obj(parse_qs(urlh.url), {
+            'state': ('state', 0),
+            'client_id': ('client', 0),
+            'protocol': ('protocol', 0),
+            'connection': ('connection', 0),
+            'scope': ('scope', 0),
+            'nonce': ('nonce', 0),
+            'response_type': ('response_type', 0),
+            'response_mode': ('response_mode', 0),
+            'redirect_uri': ('redirect_uri', 0),
+        })
```

Comment:
Maybe this can be used instead?
```suggestion
        login_info = {('client_id' if k == 'client' else k): v[0]
                      for k, v in parse_qs(urlh.url).items()}
```
It would no longer filter out any additional URL parameters that might be present. Not sure if that is desired.